airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Annotated
+
+from fastapi import Depends, HTTPException, status
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from airflow.api_fastapi.common.db.common import get_session, paginated_select
+from airflow.api_fastapi.common.parameters import (
+    QueryAssetDagIdPatternSearch,
+    QueryLimit,
+    QueryOffset,
+    QueryUriPatternSearch,
+    SortParam,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.serializers.assets import AssetCollectionResponse, AssetResponse
+from airflow.models.asset import AssetModel
+
+assets_router = AirflowRouter(tags=["Asset"], prefix="/assets")
+
+
+@assets_router.get(
+    "/",
+    responses=create_openapi_http_exception_doc([401, 403, 404]),
+)
+async def get_assets(
+    limit: QueryLimit,
+    offset: QueryOffset,
+    uri_pattern: QueryUriPatternSearch,
+    dag_ids: QueryAssetDagIdPatternSearch,
+    order_by: Annotated[
+        SortParam,
+        Depends(SortParam(["id", "uri", "created_at", "updated_at"], AssetModel).dynamic_depends()),
+    ],
+    session: Annotated[Session, Depends(get_session)],
+) -> AssetCollectionResponse:
+    """Get assets."""
+    assets_select, total_entries = paginated_select(
+        select(AssetModel),
+        filters=[uri_pattern, dag_ids],
+        order_by=order_by,
+        offset=offset,
+        limit=limit,
+        session=session,
+    )
+
+    assets = session.scalars(assets_select).all()
+    return AssetCollectionResponse(
+        assets=[AssetResponse.model_validate(x, from_attributes=True) for x in assets],
+        total_entries=total_entries,
+    )
+
+
+@assets_router.get(
+    "/{uri:path}",
+    responses=create_openapi_http_exception_doc([401, 403, 404]),
```

Comment:
Since we can pass uri in the path parameters and fastAPI encodes and decodes it, for fastAPI to be able to recognise the uri's properly, following starlette's path convertor here.

The {uri:path} portion specifies that the uri part of the path is a variable that can contain any sequence of characters, including slashes (/), making it a path parameter.

`{uri:path}`
uri is the name of the path parameter.
`:path` tells fastAPi that this parameter should capture everything in the path.

References: 
1. https://stackoverflow.com/questions/72801333/how-to-pass-url-as-a-path-parameter-to-a-fastapi-route
2. https://www.starlette.io/routing/#path-parameters